src/components/LoginPage.jsx

```diff
@@ -0,0 +1,44 @@
+import React, { useState } from 'react';
+import { useDispatch } from 'react-redux';
+import { useNavigate } from 'react-router-dom';
+import { consultUser, setUser } from '../Redux/slices/userSlice';
+import '../Style/Login.css';
+
+function LoginPage() {
+  const navigate = useNavigate();
+  const dispatch = useDispatch();
+  const [inputValue, setInputValue] = useState('');
+
+  const InputChange = (event) => {
+    setInputValue(event.target.value);
+  };
+
+  const tryLogin = async (event) => {
+    event.preventDefault();
+    const result = dispatch(consultUser(inputValue));
+    if (result) {
+      dispatch(setUser(inputValue));
+      navigate('/MainPage');
+    }
+  };
```

Comment:
Please kindly check this part of the code, when I set a non-existent username the page allows me access to the main page. The response of `result` is a promise so I suggest you return a correct response.

Reply:
Tky for your time and your comments, Solved